saleor/graphql/product/mutations/product_variant/product_variant_update.py

```diff
@@ -142,6 +146,40 @@ def post_save_action(cls, info: ResolveInfo, instance, cleaned_input):
         ).values_list("channel_id", flat=True)
         cls.call_event(mark_active_catalogue_promotion_rules_as_dirty, channel_ids)
 
+    @classmethod
+    def _save(cls, info: ResolveInfo, instance, cleaned_input, base_fields_changed):
+        new_variant = instance.pk is None
+        cls.set_track_inventory(info, instance, cleaned_input)
```

Comment:
This change will not be saved when `base_fields_changed` is False. 
IMO this code should be executed before calculating `base_fields_changed`.